bin/compile

```diff
@@ -35,16 +36,36 @@ cp -n $BP_DIR/scripts/Procfile $BUILD_DIR/Procfile
 echo "-----> Copying static.json to application slug"
 cp -n $BP_DIR/scripts/static.json $BUILD_DIR/static.json
 
-echo "-----> Installing Bower package manager"
 cd $BUILD_DIR
-npm install bower
+if [ -d "./bower_components" ]; then
+  echo "-----> Relying on cached bower_components"
+else
+  echo "DEPRECATION WARNING: This buildpack will not continue to install bower or it's dependencies in a future release. Please add `bower install` to the `postinstall` section of your application's `package.json` file. See README for more information."
 
-echo '-----> Installing Bower dependencies'
-bower install
+  if [ -x $BOWER_BIN ]; then
+    echo "-----> Using bower from package.json"
+  else
+    if [ -f "./yarn.lock" ]; then
```

Comment:
We can probably set a global to check against this once than can be re-used vs doing a file check each time for yarn.